libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/SocketChannelInterceptor.java

```diff
@@ -52,19 +52,23 @@ public static void intercept(@Advice.AllArguments Object[] args, @Origin Method
         if (args[0] instanceof InetSocketAddress address) {
             if (!AgentPolicy.isTrustedHost(address.getHostString())) {
                 final String host = address.getHostString() + ":" + address.getPort();
-
-                final SocketPermission permission = new SocketPermission(host, "connect,resolve");
+                final SocketPermission connectResolve = new SocketPermission("*", "connect,resolve");
```

Comment:
@kumargu I think this is not the right fix: the host comes from policy and has to be matched.

Reply:
you are right. I overlooked it while copying it from the policy.